src/components/integrated/OakLessonReviewItem/OakPupilLessonReviewItem.tsx

```diff
@@ -0,0 +1,174 @@
+import React from "react";
+import styled from "styled-components";
+
+import {
+  OakBox,
+  OakFlex,
+  OakIcon,
+  OakIconName,
+  OakSpan,
+} from "@/components/base";
+import { OakRoundIcon } from "@/components/ui";
+import { OakCombinedColorToken } from "@/styles";
+
+type LessonSectionName = "intro" | "starter-quiz" | "video" | "exit-quiz";
+type BaseOakLessonReviewItenProps = {
+  completed: boolean;
+};
+type QuizSectionProps = {
+  lessonSectionName: "starter-quiz" | "exit-quiz";
+  /**
+   * The number of questions in the quiz
+   */
+  numQuestions: number;
+  /**
+   * The number of questions answered correctly
+   */
+  grade: number;
+};
+
+type VideoSectionProps = {
+  lessonSectionName: "video";
+};
+type IntroSectionProps = {
+  lessonSectionName: "intro";
+};
+
+export type OakLessonReviewItemProps = BaseOakLessonReviewItenProps &
+  (IntroSectionProps | QuizSectionProps | VideoSectionProps);
+
+const StyledLessonReviewItem = styled(OakFlex)<{ completed: boolean }>`
+  outline: none;
+  text-align: initial;
+`;
+
+export const OakLessonReviewItem = (props: OakLessonReviewItemProps) => {
+  const { completed, lessonSectionName, ...rest } = props;
+  const [completedBackgroundColor, borderColor, iconBackgroundColor] =
+    pickColorsForSection(lessonSectionName);
+
+  return (
+    <StyledLessonReviewItem
+      completed={completed}
+      $gap="space-between-m"
+      $alignItems="center"
+      $background={completed ? completedBackgroundColor : "white"}
+      $ph={["inner-padding-m", "inner-padding-xl"]}
+      $pv="inner-padding-l"
+      $borderRadius="border-radius-l"
+      $borderColor={completed ? completedBackgroundColor : borderColor}
+      $ba="border-solid-l"
+      {...rest}
+    >
+      <OakRoundIcon
+        iconName={pickIconForSection(lessonSectionName)}
+        $width="all-spacing-10"
+        $height="all-spacing-10"
+        $background={iconBackgroundColor}
+      />
+      <OakFlex $flexGrow={1} $flexShrink={1} $flexDirection={"column"}>
+        <OakBox $font={["heading-6", "heading-5"]} $color={"text-primary"}>
+          {pickLabelForSection(lessonSectionName)}
+        </OakBox>
+        <OakBox $font={["body-2", "body-1"]}>
+          {pickSummaryForProgress(props)}
+        </OakBox>
+      </OakFlex>
+      {renderQuestionCounter(props)}
+    </StyledLessonReviewItem>
+  );
+};
+
+function pickSummaryForProgress(props: OakLessonReviewItemProps) {
+  switch (props.completed) {
```

Comment:
Small code style thing, but while switch is the best option for when there are multiple conditions it doesn't make sense for something as simple as this. Here I would use a simple if and return early. Then follow up with the else condition (although no need to actually use the else keyword)